src/Parser/BalanceDruid/Modules/Features/AstralPower.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import Icon from 'common/Icon';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Module from 'Parser/Core/Module';
+import SPELLS from 'common/SPELLS';
+
+class AstralPower extends Module {
+  
+  lastAstral = 0;
+  aspWasted = 0;
+
+  on_toPlayer_energize(event) {
+      for (let i = 0; i < event.classResources.length; i++) {
+          if (event.classResources[i].type === 8){
+              const maxAsP = event.classResources[i].max / 10;
+              const addedAsP = event.resourceChange;
+
+              if (this.lastAstral + addedAsP > maxAsP){
+                  this.aspWasted += this.lastAstral + addedAsP - maxAsP;
+              }
+
+              this.lastAstral = event.classResources[i].amount / 10;
+          }
+      }
+  }
+
+  on_byPlayer_cast(event) {
+      const spellId = event.ability.guid;
+      if (SPELLS.STARSURGE_MOONKIN.id !== spellId && SPELLS.STARFALL.id !== spellId && SPELLS.STARFALL_CAST.id !== spellId) {
+          return;
+      }
+      
+      for (let i = 0; i<event.classResources.length; i++) {
+          if (event.classResources[i].type === 8){
+              if (event.classResources[i].cost) {
+                  this.lastAstral = this.lastAstral - (event.classResources[i].cost / 10);
+              }
+          }
+      }
+  }
+
+  suggestions(when) {
+      when(this.aspWasted).isGreaterThan(0)
```

Comment:
If possible please try to use percentage or per minute numbers for triggers so that something like wasting 10 on a 3 minute fight can get a higher prio than wasting 10 on a 15 minute fight. I recently wrote guidelines for suggestions, here: https://github.com/MartijnHols/WoWAnalyzer/blob/master/CONTRIBUTING.SPEC.md#suggestions

Reply:
I read them last night, but I dont think they apply in this case. 
AsP is not something that should be wasted at all at any point, it's not a CD that you can forget, but a resource that needs to be controlled at all times, is the base of Moonkin DPS. 
It's controllable and really easy to not overcap, ofc the total damage is less affected on a longer fight given the same amount of AsP wasted, but overcaping 20 AsP is as big of an error in a 2min fight than in a 10m one.

Reply:
Keeping the base trigger at 0 is no problem, but the thresholds for regular/major importance are still static amounts that will trigger at their values regardless of that being after 3 minutes or 15 minutes, where I find 15 minutes to be a much more reasonable period of time to have a small amount of suboptimality.

Reply:
Let's look at this from a different PoV; if 30 AsP is 3 mil damage, this 3 mil damage comes down to 16k DPS on a 3 minute fight and 3k DPS on a 15 minute fight. This is a considerable different in my opinion.

Reply:
Yes, as I said in my comment, the consequences (DPS-wise) of overcapping 20 AsP are much more noticeables in a 1m fight than in a 12m fight.
That doesn't mean that anyone should be wasting 20 AsP, and its as big of an error. There are no situations where it would be understandable to overcap any AsP.
On top of this, the overcapping of AsP can come from many things, based on talents and # of adds, making it really difficult to stablish a limit other than "enough AsP to cast a Starsurge or a Starfall", which is what that stands for.

Reply:
I agree and understand that wasting any amount of Astral Power by overcapping (or otherwise) is an error that can and should be avoided.

I believe that suggestions should be focused on giving users the most important information before less significant information. How important information is, is loosely based around `impact on performance * ease of fixing = importance` where things with a large impact on performance and that are easy to fix get a higher importance than things with either a low impact or hard to fix. The hope is that this approach makes it so users following the higher importance suggestions first improve quicker than if they went with either low impact or difficult things.

For this suggestion my valuation is based on the following:
* **impact**: this depends on how much you wasted, but since most of the time only raw DPS matters nor total damage done I believe it depends on how much you wasted per minute. If for simplicity sake we said 20 AsP wasted = 20mil damage lost, this would be a lot of DPS in a 1 minute fight (333k DPS) and negligible in a 10 minute fight (33k DPS). Because it's negligible in a 10 minute fight, I believe the impact should be considered low in such a case.
* **ease of fixing**; let's invert this into **ease of accidentally messing up**. While wasting AsP is a big error and there are no situations where it would be understandable or acceptable to overcap any AsP, on a 10-15 minute fight where you need to concentrate for such a long period of time it's easy to let mistakes slip through. This becomes even more likely if you're focused on executing difficult mechanics over doing top DPS. Due to this I believe a low margin of error should be allowed where the suggestion isn't upgraded to a major issue.

For this discussion the important part of the equation is the *impact*.

With all this in mind I'd like to propose an approach, it shouldn't be that radically different from the current implementation.
* Calculate the AsP wasted per minute `aspWasted / this.owner.fightDuration * 1000`
* Trigger on 0 AsP wasted as it is (=minor importance)
* Upgrade to regular on 6.0 AsP wasted per minute (=30.0 per 5 minutes)
* Upgrade to major on 10.0 AsP wasted per minute (=50.0 per 5 minutes)

The numbers might need some more tuning, but I think they're fair based on what you said about the cost of Starsurge and Starfall. (The numbers would have to be multiplied by 10 to add to the suggestion code since you're using the combatlog values)

In the end this is you're spec and I'll leave the final decision up to you. Please respond here what you think and I'll accept your final choice.

Reply:
While I'd like to merge this right now I'm waiting since I'd really like to hear your decision. I'm sorry this took so long to get merged.

Reply:
Ok, I've been giving this a bit of thought, and talking with other TCs.
The main problem with AsP is that its overcapping issue is a lot worse when caused by some spells or some others (shows either just the coincidence of a couple things that can overwhelm the player, or a missunderstanding on rotationt and priorities)

So, I'm going to try and make a much more complicated - accurate - thing around it at some point (mainly checking where and why the AsP was overcapped, and drawing conclusions from it).
Since this will take time and we need to work on the numbers first, I'll go on with your idea for now (tweaking some numbers), so we have something functional as soon as possible (that is as accurate as can be), and work on that as an advanced thing once every basic module is up.

Tell me what you think, please :)

Reply:
Sounds good 👍